### PR TITLE
Replace int/uint by isize/usize in libstd/thread.rs

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
Part of #22240.

### CODE REVIEW

File: src/libstd/thread.rs [hunk 1]
```diff
@@ -805,13 +805,13 @@ mod test {
     }
 
     fn avoid_copying_the_body<F>(spawnfn: F) where F: FnOnce(Thunk<'static>) {
-        let (tx, rx) = channel::<uint>();
+        let (tx, rx) = channel::<usize>();
```
Comment (aturon) [inline src/libstd/thread.rs]: This can just as well be u32

File: src/libstd/thread.rs [hunk 2]
```diff
@@ -776,7 +776,7 @@ mod test {
 
         let (tx, rx) = channel();
 
-        fn f(i: int, tx: Sender<()>) {
+        fn f(i: isize, tx: Sender<()>) {
```
Comment (aturon) [inline src/libstd/thread.rs]: i32

File: src/libstd/thread.rs [hunk 3]
```diff
@@ -170,7 +170,7 @@ pub struct Builder {
     // A name for the thread-to-be, for identification in panic messages
     name: Option<String>,
     // The size of the stack for the spawned thread
-    stack_size: Option<uint>,
+    stack_size: Option<usize>,
```
Comment (aturon) [inline src/libstd/thread.rs]: This is fine

File: src/libstd/thread.rs [hunk 4]
```diff
@@ -283,8 +283,8 @@ impl Builder {
         // address at which our stack started).
         let main = move || {
             let something_around_the_top_of_the_stack = 1;
-            let addr = &something_around_the_top_of_the_stack as *const int;
-            let my_stack_top = addr as uint;
+            let addr = &something_around_the_top_of_the_stack as *const isize;
```
Comment (aturon) [inline src/libstd/thread.rs]: This is fine

File: src/libstd/thread.rs [hunk 5]
```diff
@@ -850,8 +850,8 @@ mod test {
         // climbing the task tree to dereference each ancestor. (See #1789)
         // (well, it would if the constant were 8000+ - I lowered it to be more
         // valgrind-friendly. try this at home, instead..!)
-        static GENERATIONS: uint = 16;
```
Comment (aturon) [inline src/libstd/thread.rs]: u32

**Comment by alexcrichton** [comment]
Thanks! r=me with a squash

Also feel free to comment on a PR whenever you update it because sadly github does not send out notifications otherwise.

**Comment by GuillaumeGomez** [comment]
@alexcrichton: I think it can be merged now. Except if @aturon sees something else which needs to be changed ?

**Comment by alexcrichton** [comment]
Could you squash the commits together as well?

File: src/libstd/thread.rs [hunk 6]
```diff
@@ -805,13 +805,13 @@ mod test {
     }
 
     fn avoid_copying_the_body<F>(spawnfn: F) where F: FnOnce(Thunk<'static>) {
-        let (tx, rx) = channel::<uint>();
+        let (tx, rx) = channel::<u32>();
 
         let x = box 1;
-        let x_in_parent = (&*x) as *const int as uint;
+        let x_in_parent = (&*x) as *const isize as usize;
```
Comment (heinrich5991) [inline src/libstd/thread.rs]: `i32` here

**Comment by GuillaumeGomez** [comment]
r=@alexcrichton

**Comment by GuillaumeGomez** [comment]
Done !

**Comment by Manishearth** [comment]
I'm getting these errors in libstd:

```
/home/manishearth/Mozilla/rust/src/libstd/panicking.rs:36:61: 36:65 warning: the `uint` type is deprecated; use `usize` or a fixed-sized integer
/home/manishearth/Mozilla/rust/src/libstd/panicking.rs:36 pub fn on_panic(obj: &(Any+Send), file: &'static str, line: uint) {
                                                                                                                      ^~~~
/home/manishearth/Mozilla/rust/src/libstd/panicking.rs:36:65: 36:65 help: add #![feature(int_uint)] to the crate attributes to silence this warning
/home/manishearth/Mozilla/rust/src/libgetopts/lib.rs:966:50: 966:66 error: failed to resolve. Could not find `uint` in `std`
/home/manishearth/Mozilla/rust/src/libgetopts/lib.rs:966     t("\nMary had a little lamb\nLittle lamb\n", ::std::uint::MAX,
                                                                                                          ^~~~~~~~~~~~~~~~
/home/manishearth/Mozilla/rust/src/libgetopts/lib.rs:966:50: 966:66 error: unresolved name `std::uint::MAX`
/home/manishearth/Mozilla/rust/src/libgetopts/lib.rs:966     t("\nMary had a little lamb\nLittle lamb\n", ::std::uint::MAX,
                                                                                                          ^~~~~~~~~~~~~~~~
error: aborting due to 2 previous errors
make: *** [x86_64-unknown-linux-gnu/stage1/test/getoptstest-x86_64-unknown-linux-gnu] Error 101
make: *** Waiting for unfinished jobs....
/home/manishearth/Mozilla/rust/src/libstd/old_io/mod.rs:1618:10: 1618:11 warning: the type parameter `T` is not constrained by the impl trait, self type, or predicates
/home/manishearth/Mozilla/rust/src/libstd/old_io/mod.rs:1618 impl<'a, T, A: ?Sized + Acceptor<T>> Iterator for IncomingConnections<'a, A> {
                                                                      ^
/home/manishearth/Mozilla/rust/src/libstd/thread.rs:818:21: 818:31 error: mismatched types:
 expected `u32`,
    found `usize`
(expected u32,
    found usize) [E0308]
/home/manishearth/Mozilla/rust/src/libstd/thread.rs:818             tx.send(x_in_child).unwrap();
                                                                            ^~~~~~~~~~
<core macros>:5:24: 5:35 error: mismatched types:
 expected `usize`,
    found `u32`
(expected usize,
    found u32) [E0308]
<core macros>:5 if ! ( ( * left_val == * right_val ) && ( * right_val == * left_val ) ) {
                                       ^~~~~~~~~~~
<core macros>:1:1: 9:39 note: in expansion of assert_eq!
/home/manishearth/Mozilla/rust/src/libstd/thread.rs:822:9: 822:45 note: expansion site
<core macros>:5:58: 5:68 error: mismatched types:
 expected `u32`,
    found `usize`
(expected u32,
    found usize) [E0308]
<core macros>:5 if ! ( ( * left_val == * right_val ) && ( * right_val == * left_val ) ) {
                                                                         ^~~~~~~~~~
<core macros>:1:1: 9:39 note: in expansion of assert_eq!
/home/manishearth/Mozilla/rust/src/libstd/thread.rs:822:9: 822:45 note: expansion site
/home/manishearth/Mozilla/rust/src/libstd/thread.rs:859:24: 859:35 error: mismatched types:
 expected `usize`,
    found `u32`
(expected usize,
    found u32) [E0308]
/home/manishearth/Mozilla/rust/src/libstd/thread.rs:859                 if x < GENERATIONS {
                                                                               ^~~~~~~~~~~
error: aborting due to 4 previous errors
make: *** [x86_64-unknown-linux-gnu/stage1/test/stdtest-x86_64-unknown-linux-gnu] Error 101
```

Could you try to make sure `check-stage1` passes?

**Comment by GuillaumeGomez** [comment]
Tests have been fixed. @alexcrichton @Manishearth

**Comment by tbu-** [comment]
@Manishearth This looks incorrect, it is casting a pointer to a `u32`. (Also it uses `isize` for an integer where it is unnecessary, but that's incorrect, just missed while replacing.)

**Comment by Manishearth** [comment]
Where? Can you file a followup bug or pull-request?

**Comment by GuillaumeGomez** [comment]
You're talking in the tests ? I didn't see that. Erf... Sorry about that...

**Comment by tbu-** [comment]
I'm working on it.

**Pull request merged**